plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableChangesFunction.java

```diff
@@ -0,0 +1,116 @@
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.VarcharType;
+import org.apache.iceberg.Table;
+
+import javax.inject.Inject;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.TypeConverter.toTrinoType;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static java.util.Objects.requireNonNull;
+
+public class IcebergTableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    private static final String START_SNAPSHOT_VAR_NAME = "START_SNAPSHOT_ID";
+    private static final String END_SNAPSHOT_VAR_NAME = "END_SNAPSHOT_ID";
+
+    private final TrinoCatalogFactory trinoCatalogFactory;
+    private final TypeManager typeManager;
+
+    @Inject
+    public IcebergTableChangesFunction(TrinoCatalogFactory trinoCatalogFactory, TypeManager typeManager)
+    {
+        super(
+                "system",
+                "table_changes",
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder()
+                                .name("SCHEMA")
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name("TABLE")
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(START_SNAPSHOT_VAR_NAME)
+                                .type(BIGINT)
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(END_SNAPSHOT_VAR_NAME)
+                                .type(BIGINT)
+                                .build()),
+                GENERIC_TABLE);
+
+        this.trinoCatalogFactory = requireNonNull(trinoCatalogFactory, "trinoCatalogFactory is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        // TODO: Is there a way to set the schema using name resolution of the function rather than putting it in "system"?
+        // TODO: Null checking
+        // TODO: Should the edges of the range be inclusive or exclusive? How to handle the boundaries like the initial Snapshot
+        String schema = ((Slice) ((ScalarArgument) arguments.get("SCHEMA")).getValue()).toStringUtf8();
+        String table = ((Slice) ((ScalarArgument) arguments.get("TABLE")).getValue()).toStringUtf8();
```

Comment:
Taking schema and table names indicates that we may want to wait for a table argument support in PTF.

Reply:
That reminds me, I wanted to move it out of the 'system' namespace